WordPress/gutenberg
# Components: ComboboxControl supports disabled items

## What?


This PR supports disabled items for the `<ComboboxControl / >` component.

## Why?


Because the `disable` option for a Combobox component is a common requirement when using it for the UI.

## How?


The `<ComboboxControl />` now accepts the `disabled` property, which is propagated to the `SuggestionsList` one.
The implementation checks if the option is disabled. If so, it does not allow selection in the respective event handler functions.

Let's remark that the disable options are not selectable (by clicking on it or pressing the `enter` key), but they are discoverable and can make the implementation a11y-friendly.

## Testing Instructions





This PR adds a new story to test the `disable` items:

1. Run the storybook 

```cli
npm run storybook:dev
```

2. Select the ComboboxControl story

```
http://localhost:50240/?path=/story/components-comboboxcontrol--with-disabled-options
```

4. Confirm the disabled items looks like disabled :smart:
5. Confirm it is not possible to select the items by clicking on them
6. Confirm it is possible to "discover" the item by pressing the UP/DOWN arrow
7. Confirm it is not possible to select the items by pressing the `ENTER` key


https://github.com/WordPress/gutenberg/assets/77539/57d42a80-3350-4a18-bc8e-be0508d84600


### Testing Instructions for Keyboard


## Screenshots or screencast
